modules/exadsBidAdapter.js

```diff
@@ -0,0 +1,511 @@
+    envParams.osName = 'Android';
+  } else if (envParams.userAgent.userAgent.match(/iPhone|iPad|iPod/i)) {
+    envParams.osName = 'iOS';
+  } else if (envParams.userAgent.userAgent.match(/Linux/i)) {
+    envParams.osName = 'Linux';
+  } else {
+    envParams.osName = 'Unknown';
+  }
+
+  let browserLanguage = navigator.language || navigator.userLanguage;
+  let acceptLanguage = browserLanguage.replace('_', '-');
+
+  envParams.language = acceptLanguage;
+
+  utils.logInfo('Domain -> ', envParams.domain);
+  utils.logInfo('Page -> ', envParams.page);
+  utils.logInfo('Lang -> ', envParams.lang);
+  utils.logInfo('OS -> ', envParams.osName);
+  utils.logInfo('User Agent -> ', envParams.userAgent);
+  utils.logInfo('Primary Language -> ', envParams.language);
+
+  return envParams;
+}
+
+export const imps = new Map();
+
+/*
+  Common mandatory parameters:
+  - endpoint
+  - userIp
+  - userIp - the minimum constraint is having the propery, empty or not
+  - zoneId
+  - partner
+  - fid
+  - siteId
+  - impressionId
+  - country
+  - mediaTypes?.banner or mediaTypes?.native or mediaTypes?.video
+
+  for native parameters
+  - assets - it should contain the img property
+
+  for video parameters
+  - mimes - it has to contain one mime type at least
+  - procols - it should contain one protocol at least
+
+*/
+function handleValidORTB2Dot4(bid) {
+  const bannerInfo = bid.mediaTypes?.banner;
+  const nativeInfo = bid.mediaTypes?.native;
+  const videoInfo = bid.mediaTypes?.video;
+  const isValid = (
+    hasValue(bid.params.endpoint) &&
+    hasValue(bid.params.userIp) &&
+    bid.params.hasOwnProperty('userIp') &&
+    hasValue(bid.params.zoneId) &&
+    hasValue(bid.params.partner) &&
+    hasValue(bid.params.fid) &&
+    hasValue(bid.params.siteId) &&
+    hasValue(bid.params.impressionId) &&
+    hasValue(bid.params.country) &&
+    hasValue(bid.params.country.length > 0) &&
+    ((!hasValue(bid.params.bcat) ||
+      bid.params.bcat.length > 0)) &&
+    ((!hasValue(bid.params.badv) ||
+        bid.params.badv.length > 0)) &&
+    (bannerInfo || nativeInfo || videoInfo) &&
+    (nativeInfo ? bid.params.native &&
+      nativeInfo.ortb.assets &&
+      nativeInfo.ortb.assets.some(asset => !!asset.img) : true) &&
+    (videoInfo ? (videoInfo.mimes &&
+      videoInfo.mimes.length > 0 &&
+      videoInfo.protocols &&
+      videoInfo.protocols.length > 0) : true));
+  if (!isValid) {
+    utils.logError('Validation Error');
+  }
+
+  return isValid;
+}
+
+function hasValue(value) {
+  return (
+    value !== undefined &&
+    value !== null
+  );
+}
+
+function getGdprConsentChoice(bidderRequest) {
+  const hasGdprConsent =
+    hasValue(bidderRequest) &&
+    hasValue(bidderRequest.gdprConsent);
+
+  if (hasGdprConsent) {
+    return bidderRequest.gdprConsent;
+  }
+
+  return null;
+}
+
+export const spec = {
+  aliases: ['exads'], // short code
+  supportedMediaTypes: [BANNER, NATIVE, VIDEO],
```

Comment:
You are missing your `GVL_ID` here ( 1084 )

Reply:
Yes, you are good. Just put the GVL_ID.